pkg/resources/secret_with_basic_authentication.go

```diff
@@ -0,0 +1,163 @@
+
+	id := sdk.NewSchemaObjectIdentifier(commonCreate.database, commonCreate.schema, commonCreate.name)
+
+	usernameString := d.Get("username").(string)
+	passwordString := d.Get("password").(string)
+
+	request := sdk.NewCreateWithBasicAuthenticationSecretRequest(id, usernameString, passwordString)
+	if v, ok := d.GetOk("comment"); ok {
+		request.WithComment(v.(string))
+	}
+
+	err := client.Secrets.CreateWithBasicAuthentication(ctx, request)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(helpers.EncodeResourceIdentifier(id))
+
+	return ReadContextSecretWithBasicAuthentication(ctx, d, meta)
+}
+
+func ReadContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	secret, err := client.Secrets.ShowByID(ctx, id)
+	if secret == nil || err != nil {
+		if errors.Is(err, sdk.ErrObjectNotFound) {
+			d.SetId("")
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Warning,
+					Summary:  "Failed to retrieve secret. Target object not found. Marking the resource as removed.",
+				},
+			}
+		}
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to retrieve secret.",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err),
+			},
+		}
+	}
+	secretDescription, err := client.Secrets.Describe(ctx, id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if err := handleSecretRead(d, id, secret, secretDescription); err != nil {
+		return diag.FromErr(err)
+	}
+	if err = d.Set("username", secretDescription.Username); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func UpdateContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if request := handleSecretUpdate(id, d); request != nil {
+		if err := client.Secrets.Alter(ctx, request); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	if d.HasChange("username") {
+		username := d.Get("username").(string)
+
+		request := sdk.NewAlterSecretRequest(id)
+		setRequest := sdk.NewSetForBasicAuthenticationRequest().WithUsername(username)
+		request.WithSet(*sdk.NewSecretSetRequest().WithSetForBasicAuthentication(*setRequest))
+
+		if err := client.Secrets.Alter(ctx, request); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	if d.HasChange("password") {
+		password := d.Get("password").(string)
+
+		request := sdk.NewAlterSecretRequest(id)
+		setRequest := sdk.NewSetForBasicAuthenticationRequest().WithPassword(password)
+		request.WithSet(*sdk.NewSecretSetRequest().WithSetForBasicAuthentication(*setRequest))
+
+		if err := client.Secrets.Alter(ctx, request); err != nil {
```

Comment:
We rather prefer to have as few SQL calls as possible. I can see in the documentation that you can set username, password, and comment at once. Knowing that you can create alter request at the top and at the bottom make a comparison if anything was set (there are some examples in codebase e.g. resource_monitor except there's no request used there, but plain SQL struct. You can achieve the same with alter request).

Reply:
You would have to split comment changes into set request that would handle set comment, username, password and unset request that would handle unset comment (so still 2 SQL calls at max, but that's still smaller than before).

Reply:
I did exactly that followin what have been done in `api_authentication_integration` what @sfc-gh-jmichalak has mentioned in his comments